### ISSUE TITLE
Why does Yarp return 502 status code when a client disconnects?

### ISSUE THREAD

**Issue by adriaanp** [microsoft/reverse-proxy#2019]
502 (Bad Gateway) is usually returned when the reverse proxy encounters an error while trying to forward the request to the backend server, but in the case where a client disconnects should it not set the status code to 499?

**Comment by MihaZupan** [comment]
By "return", are you just referring to the values you see in logs? By definition, the client won't see the value.

**Comment by adriaanp** [comment]
Yes, that's correct, the client won't see the value, I meant in logs and observability tools like new relic.

**Comment by Tratcher** [comment]
499 is a non-standard code that NGinx made up. It being a defacto standard might be sufficient cause to use it though.

In general, you should look at the ForwarderError for a more detailed explanation.
https://github.com/microsoft/reverse-proxy/blob/79a7b806688311fd2cca2ca67dc7cc418c7925ab/src/ReverseProxy/Forwarder/ForwarderError.cs#L39

**Comment by Tratcher** [comment]
We're starting to use 499 in AspNetCore for this situation as well: https://github.com/dotnet/aspnetcore/pull/46330
